components/external_child_process/child_monitor.cc

```diff
@@ -0,0 +1,171 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/external_child_process/child_monitor.h"
+
+#if defined(OS_POSIX)
+#include <errno.h>
+#include <fcntl.h>
+#include <signal.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#endif
+
+#include <utility>
+
+#include "base/bind_post_task.h"
+#include "base/logging.h"
+#include "base/process/kill.h"
+#include "base/single_thread_task_runner.h"
+#include "base/task/post_task.h"
+#include "base/threading/thread_task_runner_handle.h"
+
+namespace brave {
+
+namespace {
+
+#if defined(OS_POSIX)
+int pipehack[2];
+
+static void SIGCHLDHandler(int signo) {
+  int error = errno;
+  char ch = 0;
+  (void)write(pipehack[1], &ch, 1);
+  errno = error;
+}
+
+static void SetupPipeHack() {
+  if (pipe(pipehack) == -1)
+    VLOG(0) << "pipehack errno:" << errno;
+
+  int flags;
+  for (size_t i = 0; i < 2; ++i) {
```

Comment:
since there's only two items, a read end and a write end, i don't think a loop adds much value here. you can pull this out into a function and call that twice. you can add a paramter `bool: write_end` and then change `if i==1 { .. }` (not very clear) to `if write_end { .. }`

Reply:
fixed in fcf4957f689f39e8f09a0c9aa56494935fcbfd7d